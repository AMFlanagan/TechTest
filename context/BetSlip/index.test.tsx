import { render } from '@testing-library/react'
import { BetSlipProvider, bestSlipReducer, ActionTypes, IBetSlipEntry } from '.'

describe('BetSlipContext', () => {
  it('renders as expected', () => {
    const component = <div>test</div>

    const { asFragment } = render(
      <BetSlipProvider>{component}</BetSlipProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('betSlipReducer', () => {
  it('adds a bet', () => {
    const state = {
      bets: [
        {
          eventId: 1,
          outcomeId: 1,
        } as IBetSlipEntry,
      ],
    }

    const action = {
      type: ActionTypes.CLEAR_ALL,
    }

    const newState = bestSlipReducer(state, action)

    expect(newState).toEqual({
      bets: [],
    })
  })

  it('adds a bet', () => {
    const state = {
      bets: [],
    }

    const action = {
      type: ActionTypes.ADD,
      payload: {
        eventId: 1,
        outcomeId: 1,
      },
    }

    const newState = bestSlipReducer(state, action)

    expect(newState).toEqual({
      bets: [
        {
          eventId: 1,
          outcomeId: 1,
        },
      ],
    })
  })

  it('removes a bet', () => {
    const state = {
      bets: [
        {
          eventId: 1,
          outcomeId: 1,
        } as IBetSlipEntry,
        {
          eventId: 2,
          outcomeId: 2,
        } as IBetSlipEntry,
      ],
    }

    const action = {
      type: ActionTypes.REMOVE,
      payload: {
        eventId: 1,
        outcomeId: 1,
      },
    }

    const newState = bestSlipReducer(state, action)

    expect(newState).toEqual({
      bets: [
        {
          eventId: 2,
          outcomeId: 2,
        },
      ],
    })
  })

  it('updates a stake', () => {
    const state = {
      bets: [
        {
          eventId: 1,
          outcomeId: 1,
        } as IBetSlipEntry,
        {
          eventId: 2,
          outcomeId: 2,
        } as IBetSlipEntry,
      ],
    }

    const action = {
      type: ActionTypes.UPDATE,
      payload: {
        eventId: 1,
        outcomeId: 1,
        stake: 1,
      },
    }

    const newState = bestSlipReducer(state, action)

    expect(newState).toEqual({
      bets: [
        {
          eventId: 1,
          outcomeId: 1,
          stake: 1,
        },
        {
          eventId: 2,
          outcomeId: 2,
        },
      ],
    })
  })
})
