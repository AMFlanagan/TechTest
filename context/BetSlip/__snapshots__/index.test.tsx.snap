// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetSlipContext renders as expected 1`] = `
<DocumentFragment>
  <div>
    test
  </div>
</DocumentFragment>
`;
